arch/x86/kvm/pkvm/memory.h

```diff
@@ -6,13 +6,16 @@
 #include <linux/range.h>
 #include <vdso/limits.h>
 #include <asm/page.h>
+#include "mem_protect.h"
 
 #define __pkvm_pa		__pa
 #define __pkvm_va		__va
 
 struct pkvm_page {
 	unsigned short refcount;
 	unsigned short order;
+	/* Store host memory page state. */
+	enum pkvm_page_state host_state;
```

Comment:
In pKVM-ARM the layout of this structure is:

```
struct hyp_page {
	unsigned short refcount;
	u8 order;

	/* Host (non-meta) state. Guarded by the host stage-2 lock. */
	enum pkvm_page_state host_state : 8;

	u32 host_share_guest_count;
};
```

so refcount + order + host_state consume only 4 bytes instead of 8 bytes.

We could do the same (so that e.g. on a machine with 32GB of RAM, pkvm's vmemmap would only require 32MB instead of 64MB)?

Reply:
Makes sense. I will reduce the size to 4 bytes.

Reply:
As we have a macro PKVM_PAGE_STATE_BITS, probably can leverage that instead of hardware code with 8?

```
  struct pkvm_page {
          unsigned short refcount;
          u8 order;

          /* Store host memory page state. */
          enum pkvm_page_state host_state: PKVM_PAGE_STATE_BITS;
  };
```